### ISSUE TITLE
tf.keras.ops.exp2 returns infinite gradient although the float64 derivative is finite

### ISSUE THREAD

**Issue by ALinrunrun** [tensorflow/tensorflow#126627]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf 2.21.0

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 22.04

### Python version

Python 3.13

### Current behavior?

`tf.keras.ops.exp2` returns an infinite first derivative even though the correctly rounded float64 derivative is finite and normal.

For `f(x) = 2**x`, the derivative is `ln(2) * 2**x`. At `x = 1024.0`, the forward value overflows to `inf` in float64, but the derivative is still representable as a finite float64 value. A high-precision reference rounded directly to float64 gives approximately `1.2460659279417838e+308`.

However, TensorFlow autodiff returns `inf` for this derivative.

Expected behavior?

The first derivative of `tf.keras.ops.exp2` at `x = 1024.0` should be close to `1.2460659279417838e+308`, not `inf`.

### Standalone code to reproduce the issue

```shell
import os
os.environ.setdefault("CUDA_VISIBLE_DEVICES", "-1")
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

import numpy as np
import tensorflow as tf

x_value = 1024.0
expected = 1.2460659279417838e308
rtol = 1e-6

x = tf.Variable(x_value, dtype=tf.float64)

with tf.GradientTape() as tape:
    y = tf.keras.ops.exp2(x)

g = tape.gradient(y, x)

actual = float(g)
rel_err = abs(actual - expected) / abs(expected)

print(f"x: {x_value!r}")
print(f"forward value: {float(y)!r}")
print(f"tensorflow gradient: {actual!r}")
print(f"expected gradient: {expected!r}")
print(f"relative error: {rel_err}")

if not np.isfinite(actual) and np.isfinite(expected):
    print("BUG REPRODUCED: tf.keras.ops.exp2 gradient is infinite although the float64 result is finite")
else:
    print("not reproduced")
```

### Relevant log output

```shell
x: 1024.0
forward value: inf
tensorflow gradient: inf
expected gradient: 1.2460659279417838e+308
relative error: inf
BUG REPRODUCED: tf.keras.ops.exp2 gradient is infinite although the float64 result is finite
```

**Comment by Venkat6871** [comment]
Hi @ALinrunrun ,
Thank you for reporting this issue with tf.keras.ops.exp2 returning an infinite gradient when the float64 derivative is expected to be finite. Your detailed description, clear reproduction steps, and the provided code snippet are very helpful. We will investigate this from our side and provide an update.
